### ISSUE TITLE
Differentiate between type definitions and usages

### ISSUE THREAD

**Issue by p00f** [nvim-treesitter/nvim-treesitter#5005]
**Is your feature request related to a problem? Please describe.**
Type definitions and usages should be differentiated in highlighting, for example highlighting usages less strongly (or not at all). Currently `@type.definition` is only used for type aliases.

**Describe the solution you'd like**
`@type.definition` should be changed to capture type definitions, like struct/class definitions.

Additionally, the current usages of `@type.definition` should be changed to `@type.alias`, as they are aliases (though this is a breaking change)

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by p00f** [comment]
@theHamsta @savq

**Comment by amaanq** [comment]
To extend on adding type differentiation, `@interface` would be a nice capture to add as well (C++ virtual classes, Rust/Typescript/Java interfaces),

**Comment by p00f** [comment]
@casouri

**Comment by p00f** [comment]
> Additionally, the current usages of `@type.definition` should be changed to `@type.alias`, as they are aliases (though this is a breaking change)

s/should/can - I don't feel too strongly about this since an alias is also a new type

**Comment by casouri** [comment]
Yes. AFAIK Emacs major modes fontify types uniformly, even though functions and variables distinguish between definition and usage.

**Comment by theHamsta** [comment]
I think the differentiation between definition and usage would be a good fit when we can support it more generally (e.g. also for parameters via locals). It would be good to offer definition vs usage in a uniform way. As far as I'm aware we're already differentating function definition vs call. So it could be possible to do it for types as well. Would still be glad if we could think whether the concept definition/usage could be streamlined among all captures that we might introduce in future
